include/SFML/System/Clock.hpp

```diff
@@ -30,10 +30,59 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/System/Export.hpp>
 #include <SFML/System/Time.hpp>
+#include <chrono>
+#include <ratio>
+#include <type_traits>
+
+#ifdef SFML_SYSTEM_ANDROID
+#include <SFML/System/Android/SuspendAwareClock.hpp>
+#endif
```

Comment:
The reason the Android build is failing, is because you're including a "private" header in public header. The SuspendAwareClock is located in the `src` directory, which is not installed with the SFML distribution, yet the System header in the public `include` directory, tries to include it.
Thus, when building the Android example and using the public SFML headers, it fails, as the private header can't be found.

The preferred solution would be to use the pImpl pattern to hide the Android implementation.

Reply:
Pimpl would be good at hiding implementation details at run-time. However, the choice of the clock is done at compile-time.

It would be very inelegant to move it to run-time unless you have a better solution to that.

Reply:
@reconndev: I don't understand, you could do something like this:

```cpp
#ifdef SFML_SYSTEM_ANDROID
class SuspendAwareClock;
#endif

// ...

#if defined(SFML_SYSTEM_ANDROID) && defined(SFML_ANDROID_USE_SUSPEND_AWARE_CLOCK)
    using MostSuitableClock = SuspendAwareClock;
#else
    using MostSuitableClock = std::conditional_t<
        std::chrono::high_resolution_clock::is_steady,
        std::chrono::high_resolution_clock,
        std::chrono::steady_clock>;
#endif

// ...

using ClockImpl = std::unique_ptr<priv::MostSuitableClock>;
```

You would have to expose inner type aliases such as `time_point` though, so I'm not sure how worth it this change would be.

Reply:
You have to write more code to see the problem:
https://godbolt.org/z/fGEEsojfP

Reply:
That's what I meant with *"You would have to expose inner type aliases such as time_point though, so I'm not sure how worth it this change would be."*, so I don't think it's worth performing this change.